cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -294,48 +295,62 @@ sanityCheckElaboratedPackage ElaboratedConfiguredPackage{..}
 rebuildProjectConfig :: Verbosity
                      -> DistDirLayout
                      -> ProjectConfig
-                     -> IO (ProjectConfig,
-                            [PackageSpecifier UnresolvedSourcePackage])
-rebuildProjectConfig verbosity
-                     distDirLayout@DistDirLayout {
-                       distProjectRootDirectory,
-                       distDirectory,
-                       distProjectCacheFile,
-                       distProjectCacheDirectory
-                     }
-                     cliConfig = do
+                     -> IO
+                          ( ProjectConfig
+                          , [PackageSpecifier UnresolvedSourcePackage]
+                          )
+rebuildProjectConfig verbosity distDirLayout cliConfig = do
 
     (projectConfig, localPackages) <-
-      runRebuild distProjectRootDirectory $
-      rerunIfChanged verbosity fileMonitorProjectConfig () $ do
-
-        projectConfig <- phaseReadProjectConfig
-        localPackages <- phaseReadLocalPackages projectConfig
-        return (projectConfig, localPackages)
+      runRebuild distProjectRootDirectory
+      $ rerunIfChanged verbosity
+                       fileMonitorProjectConfig
+                       (projectConfigConfigFile, distProjectFile "")
+      $ do
+          liftIO $ info verbosity "Project settings changed, reconfiguring..."
+          projectConfig <- phaseReadProjectConfig
+          localPackages <- phaseReadLocalPackages projectConfig
+          return (projectConfig, localPackages)
+
+    when (verbosity >= verbose) $ do
+      monitorFileList <- getFileMonitorSimpleFileList fileMonitorProjectConfig
+      info verbosity
+        $ unlines
+        $ ("this build was affected by the following (project) config files:" :)
+        $ filter (\x -> not (".cabal" `isSuffixOf` x) && x /= ".")
```

Comment:
Filtering out only cabal files might lead to a lot of false-posives in this list. Consider the support for local tarballs, which adds a file monitor to a `.tar.gz` file which will count as "config file" here.

Reply:
raah that's a good point, but I don't see any elegant to solution for this without creating some kind of duplication or implication what config files we have. Perhaps this should just not do any filtering and not pretend it only contains config files, but "configuration input stuff"?

Reply:
Well, we have `distProjectFile` in scope, we could do something like this:

filterM doesFileExist $ map distProjectFile ["", "local", "freeze"]

Reply:
Or better: we have `projectConfigProvenance` which should hold all the paths from the involved configuration files.